src/ddmd/traits.d

```diff
@@ -54,53 +54,1012 @@ struct PushAttributes
 
 extern (C++) __gshared StringTable traitsStringTable;
 
+/**
```

Comment:
Made one more pass, a thought: this exposes a bunch of nicely-named symbols. Can we make them private and still generate the documentation?

Reply:
I don't think so (without bug hacks/work), but DMD != Phobos, so having public symbols shouldn't a big problem. Currently the only users are GCC + LDC and they can't use this without C++ interfaces. Also we are stuck with these traits nearly forever, so I doubt that worrying about the deprecation of a public symbol is a worry here. Deprecating a trait from the language is a different pair of shoes though.